packages/libs/error-stack/src/display.rs

```diff
@@ -0,0 +1,337 @@
+//! Implementation of display, which can be either fancy or "normal", depending
+//! on the type color and non ascii character can be used.
+//!
+//! This is inspired by [miette](https://docs.rs/miette/latest/miette/index.html)
+
+#[cfg(not(feature = "std"))]
+use alloc::string::ToString;
+use alloc::{borrow::ToOwned, format, string::String, vec, vec::Vec};
+
+use crate::{AttachmentKind, Frame, FrameKind, Report};
+
+#[cfg(feature = "fancy")]
+const VERTICAL: &str = "\x1b[31m│\x1b[0m   ";
+#[cfg(not(feature = "fancy"))]
+const VERTICAL: &str = "|   ";
+
+#[cfg(feature = "fancy")]
+const ENTRY: &str = "\x1b[31m├─▶\x1b[0m ";
+#[cfg(not(feature = "fancy"))]
+const ENTRY: &str = "|-> ";
+
+#[cfg(feature = "fancy")]
+const ENTRY_END: &str = "\x1b[31m╰─▶\x1b[0m ";
+#[cfg(not(feature = "fancy"))]
+const ENTRY_END: &str = "\\-> ";
+
+const SPACE: &str = "    ";
+
+#[cfg(all(nightly, feature = "std"))]
+fn backtrace<'a>(frame: &'a Frame, bt: &mut Vec<&'a std::backtrace::Backtrace>) -> Option<String> {
+    if let Some(backtrace) = frame.request_ref::<std::backtrace::Backtrace>() {
+        if matches!(
+            backtrace.status(),
+            std::backtrace::BacktraceStatus::Unsupported
+                | std::backtrace::BacktraceStatus::Disabled
+        ) {
+            return None;
+        }
```

Comment:
Yes, well it only makes them hard in the context of doctests (because we cannot use the test_helpers crate), for this purpose I created the functions: `expect_messages` and `expect_count`, which automatically inject (if supported and needed) the `Opaque` frames in between. This makes the code clean and is pretty straightforward.

Reply:
It may make sense to drop the assertions in the doc tests and instead write proper tests. This is a task I'm happy to pick as a follow-up if you don't want to do the additional work 🙂

Reply:
Deferred to doctest update PR